client/extensions/woocommerce/app/settings/email/email-settings/index.js

```diff
@@ -95,18 +106,29 @@ class Settings extends React.Component {
 		this.fetchSettings( this.props );
 	};
 
-	componentWillReceiveProps = newProps => {
-		if ( newProps.siteId !== this.props.siteId ) {
-			this.fetchSettings( newProps );
+	componentWillReceiveProps( nextProps ) {
+		if ( nextProps.siteId !== this.props.siteId ) {
+			this.fetchSettings( nextProps );
 		}
-	};
 
-	notificationsToggle = () => {
-		undefined;
+		// Save settings request
+		if ( ( ! this.props.saveSettingsRequest ) && nextProps.saveSettingsRequest ) {
+			nextProps.loaded && nextProps.submit( nextProps.siteId, nextProps.settings );
+		}
+
+		// Settings save request finished.
+		if ( ( ! nextProps.isSaving ) && this.props.isSaving ) {
+			if ( nextProps.submitError ) {
+				nextProps.errorNotice( translate( 'There was a problem saving the email settings. Please try again.' ) );
+			} else {
+				nextProps.successNotice( translate( 'Email settings saved.' ), { duration: 4000 } );
```

Comment:
Hi! I've found a possible matching string that has already been translated **95** times:
`translate( 'Settings saved.' )` *ES Score: 7*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/email-settings-ui-save)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).